dhis-2/dhis-api/src/main/java/org/hisp/dhis/common/DimensionalObjectUtils.java

```diff
@@ -192,7 +192,7 @@ public static String getDimensionFromParam( String param )
 
         if ( param.split( DIMENSION_NAME_SEP ).length > 1 )
         {
-            return new ArrayList<>( Arrays.asList( param.split( DIMENSION_NAME_SEP )[1].split( OPTION_SEP ) ) );
+            return Arrays.asList( param.substring( param.indexOf( DIMENSION_NAME_SEP ) + 1 ).split( OPTION_SEP ) );
```

Comment:
Could you extract this `param.substring( param.indexOf( DIMENSION_NAME_SEP ) + 1 ).split( OPTION_SEP )` into a variable, so we clearly know what it refers to?

Reply:
> Hi @gnespolino, thanks for these changes. I left one comment in the code and I have another one:
> 
> I saw in your comment that this will support: `pe:LAST_WEEK;TODAY;YESTERDAY:LAST_UPDATED`
> 
> Will we also support something like: `pe:LAST_WEEK;TODAY;YESTERDAY:LAST_UPDATED;LAST_MONTH:INCIDENT_DATE` or `pe:LAST_WEEK;TODAY;YESTERDAY:LAST_UPDATED;DAILY:20220128` ?
> 
> I'm not sure if this is part of the requirements...

Yes we do support multiple, but the client should always pass in the form
`pe:LAST_WEEK;TODAY&lastUpdated=YESTERDAY`
in other words, the `pe:...` format, from client point of view, is kept exactly as it was before this PR.
Internally, to threat new time fields as periods, we convert the `lastUpdated=YESTERDAY` by appending `;YESTERDAY:LASTUPDATED` to existing `pe`.

Reply:
> Could you extract this `param.substring( param.indexOf( DIMENSION_NAME_SEP ) + 1 ).split( OPTION_SEP )` into a variable, so we clearly know what it refers to?

sure